src/meep/vec.hpp

```diff
@@ -186,6 +186,70 @@ component first_field_component(field_type ft);
          loop_ibound++)                                                                            \
   LOOP_OVER_IVECS(gv, loop_notowned_is, loop_notowned_ie, idx)
 
+// The following work identically to the LOOP_* macros above,
+// but are optimized for shared memory parallelism. This
+// is really only implemented in step_generic and code
+// generated from step_generic
+
+/* for the parallel implementation, we
+introduce 2 dummy loops, one at the begininnging
+and one at the end, in order to "trick" openMP
+to allow us to define our localy variables
+without having to change any other code
+in the main codebase. We can proceed to do
+a collapse over all three main loops. */
+
+#define CHUNK_OPENMP _Pragma("omp parallel for")
+
+// the most generic use case where the user
+// can specify a custom clause
+#define PLOOP_OVER_IVECS_C(gv, is, ie, idx, clause)			                                                     \
+_Pragma("unroll(1)") \
```

Comment:
@smartalecH, note that these `#pragma unroll` directives affect the compiler even without openmp … did they actually improve performance or were they just an experiment?

It seems that you actually weren't using them, since in your branch they were written as `_Pragma("unroll(1")` (missing a close paren), which is an error and presumably was ignored by the compiler.

Maybe we should just delete them?

Reply:
The idea was to unroll the overhead of the dummy loop (just 1 iteration) that I used to wrap everything (see comment above code). 

I didn't realize that I was missing a paren, so you can probably safely omit it. The compiler hopefully optimizes out loops of 1 iteration anyway...

Reply:
I just did a quick check in Julia with LLVM (since Julia lets me interactively examine compiler output) and it seems that it can indeed optimize out loops with a small constant number of iterations.   It should be the same in C++, and hopefully also with gcc.